param/__init__.py

```diff
@@ -27,7 +27,7 @@
 
 from .parameterized import logging_level # pyflakes:ignore (needed for eval)
 from .parameterized import shared_parameters # pyflakes:ignore (needed for eval)
-
+from .io import *
```

Comment:
My opinion on * imports is that they are okay if the module being imported uses `__all__`, but if not using `__all__` then * imports are a bad idea.  I can see how they lead to less clear code and am happy to defer - I'll fix them.  I will try to get this new material of the PR as small as possible.  Are you sure we want to get rid of the new `io.py` file entirely?  It seems like a logical separation of concerns to put IO assets in a separate file.